### PR TITLE
Clean UX so it is more clear that a pass is required

### LINKED ISSUE #179: Make required private password entry on 1st rescan more clear
After the user starts a fresh wallet or created an existing wallet from seed, the private password is required for properly rescan.

The prompt that is currently used is very hard for the user to discern that it is required that they need to do anything.  

Future, more complicated, fix would probably require shuffling stuff around to change when the password is asked.  

Easy fix is to not use logger to use as the prompt.  Something like:

`Enter private password:

-----------------------------`

### PR DESCRIPTION
**Pull request by alexlyp**
Fixes #179

### CODE REVIEW

File: dcrwallet.go [hunk 1]
```diff
@@ -188,6 +188,7 @@ func startPromptPass(w *wallet.Wallet) {
 			"This will complete syncing of the wallet accounts " +
 			"and then leave your wallet unlocked. You may relock " +
 			"wallet after by calling 'walletlock' through the RPC.")
+		backendLog.Flush()
```
Comment (jrick) [inline dcrwallet.go]: I would move this to before where the actual prompt occurs.  The prompt is run in a loop and code after the prompt in that same loop may write to the logger, so it needs to be flushed again.

**Comment by jcvernaleo** [comment]
Looks good to me.
tACK

**Comment by davecgh** [comment]
Looks good.  This should be a lot harder to miss.

**Comment by chappjc** [comment]
But this turns out to be really confusing if you start your wallet with --prompt pass and it's **not** the first time you've started it.

**Comment by davecgh** [comment]
I agree that it's not the best solution, but it solves the immediate concern where it's super easy to miss that you need to type something.  There have been several users that completely missed it and then were wondering why it wasn't responding to RPC requests.

Personally, I don't like it prompting at all without the user specifically directing it to via a flag.

Also, I proposed that the RPC server should go into an "I need to be unlocked" mode where any requests would return an appropriate error like "-X: RPC server needs to be unlocked with the `walletpass` command in order to complete initial sync" whenever that is the case.

That is a much larger change however.

**Comment by chappjc** [comment]
I agree that unsolicited prompting is somewhat undesirable, and I said so before (https://github.com/decred/dcrwallet/pull/142#issuecomment-204099902).  But I guess it's needed on the first start.

The "I need to be unlocked" mode sounds good to me.

**Pull request merged**